proposals/nnnn-screen-manager-alerts.md

```diff
@@ -0,0 +1,231 @@
+*/
+- (void)cancel;
+
+@end
+```
+
+##### Java
+```java
+public class AlertView {
+    private Integer defaultTimeout = 5;
+
+    private String text, secondaryText, tertiaryText;
+    private Integer timeout;
+    private AlertAudioData audio;
+    private Bool showWaitIndicator;
+    private List<SoftButtonObject> softButtons;
+    private SdlArtwork icon;
+
+    // All vars have get/set as well
+
+    public AlertView(@NonNull String text, @NonNull List<SoftButtonObject> softButtons)
+    public AlertView(@Nullable String text, @Nullable String secondaryText, @Nullable String tertiaryText, @Nullable Integer timeout, @NonNull Bool showWaitIndicator, @Nullable AlertAudioData audio, @Nullable List<SoftButtonObject> softButtons, @Nullable SdlArtwork icon)
+
+    public void cancel()
+}
+```
+
+And then the additions to the screen manager public API itself to present the alert.
+
+##### iOS
+```objc
+@interface SDLScreenManager: NSObject
+// Everything already there
+
+/**
+ Present the alert on the screen.
+ 
+ If the alert contains an audio indication with a file that needs to be uploaded, it will be uploaded before presenting the alert. If the alert contains soft buttons with images, they will be uploaded before presenting the alert. If the alert contains an icon, that will be uploaded before presenting the alert.
+ 
+ The handler will be called when the alert either dismisses from the screen or it has failed to present. If the error value in the handler is present, then the alert failed to appear, if not, then the alert dismissed without error. The error will contain `userInfo` with information on how long to wait before retrying.
+ */
+- (void)presentAlert:(SDLAlertView *)alert withCompletionHandler:(nullable SDLScreenManagerUpdateCompletionHandler)handler;
+
+@end
+```
+
+##### Java
+```java
+public class BaseScreenManager {
+    public void presentAlert(AlertView alert, CompletionListener listener)
+}
+```
+
+### JavaScript Suite APIs
+Due to the size of the iOS APIs and the similarity between the iOS, Java Suite and eventual JavaScript Suite APIs, this proposal does not present the public APIs of the JavaScript Suite APIs – especially because the JavaScript Suite APIs do not currently have a screen manager layer. The JavaScript Suite APIs should mirror the iOS and Java Suite API appropriately and is up to the maintainers' discretion. However, if any changes needed to be made such that they impacted the iOS / Java Suite API (such as the alteration, addition, or removal of a method or property), then a proposal revision would be needed.
+
+### Additional Implementation Notes
+- The internal alert manager will observe the `AlertResponse` to know when the alert has finished presenting, and then call the `completionHandler`.
+- The internal alert manager will always send the alert, even if the system context is not MAIN. If the `AlertResponse` returns an failure to present, it will call the `completionHandler` with the error.
```

Comment:
```suggestion
- The internal alert manager will always send the alert, even if the system context is not MAIN. If the `AlertResponse` returns a failure to present, it will call the `completionHandler` with the error.
```